Add documentation on how to securely archive data of a dismissed project (#267)
pagopa/dx: website/docs/infrastructure/archive-data.md

# Archiving Data from Decommissioned Projects

## Overview

> When a project is decommissioned, it's crucial to maintain data for legal and auditing purposes while minimizing storage costs. This guide provides a comprehensive approach to data archiving with a focus on Azure storage solutions.

Reviewer: ```suggestion
When a project is decommissioned, it is important to strike a balance between data retention for legal and auditing purposes and storage cost optimization. Depending on the context and specific requirements, adopting low-cost storage solutions can be considered while ensuring access to the necessary information for potential future audits.. This guide provides a comprehensive approach to data archiving with a focus on Azure storage solutions.
```

I would highlight that the requirements vary from project to project.

## Best Practices for Data Archiving

### Recommended Storage Account Configuration

The optimal storage solution balances flexibility, support, and cost-effectiveness with the following characteristics:

> 1. **Blob Storage Configuration**

Reviewer: ```suggestion
```

>    - Use Archive storage tier
>    - Disable internet access
>    - Enforce authentication via Entra ID (disable access keys)
>    - Implement legal hold policies
>    - Configure multi-AZ replication in the primary region
>    - Set up single-AZ replication in a secondary region
>    - Enable object replication between regions
>    - Apply read-only lock

Reviewer: ```suggestion
- Use Archive storage tier
- Disable internet access
- Enforce authentication via Entra ID (disable access keys)
- Implement legal hold policies
- Configure multi-AZ replication in the primary region
- Set up single-AZ replication in a secondary region
- Enable object replication between regions
- Apply read-only lock
```

### Sample Terraform Configuration

```hcl

> resource "azurerm_storage_account" "backup_primary" {

Reviewer: what about adding an `azure_storage_archive` module that _eventually_ extends [azure_storage_account](https://github.com/pagopa/dx/tree/main/infra/modules/azure_storage_account) ? or a special tier ('archive') in the existings `azure_storage_account` module?

Author: We should add support of entities as well, but we can think about it

  name                = replace("${local.project}backupst01", "-", "")
  resource_group_name = azurerm_resource_group.rg_itn_01.name
  location            = local.location

  account_kind             = "StorageV2"
  account_tier             = "Standard"
  account_replication_type = "ZRS"
  access_tier              = "Cool"

  public_network_access_enabled = false

  shared_access_key_enabled       = false
  default_to_oauth_authentication = true

  blob_properties {
    versioning_enabled       = true
    change_feed_enabled      = true
    last_access_time_enabled = true

    delete_retention_policy {
      days = 7
    }

    restore_policy {
      days = 5
    }

    container_delete_retention_policy {
      days = 10
    }
  }
}
```

> > **Note:** 

Reviewer: ```suggestion
:::note

```

> > - Manually set the `access_tier` to `Archive` via Azure Portal after backup completion
> > - Set `public_network_access_enabled` to `false` only after backup is complete
> 

Reviewer: ```suggestion
- Manually set the `access_tier` to `Archive` via Azure Portal after backup completion
- Set `public_network_access_enabled` to `false` only after backup is complete

:::
```

> ### Secondary Region Options
> - Germany West Central (GWC)
> - Spain Central (SPC)

Reviewer: ```suggestion
### Secondary Region Options

- Germany West Central (GWC)
- Spain Central (SPC)
```

## Data Migration Strategies

> ### 1. Migrating from Cosmos DB

Reviewer: ```suggestion
### Migrating from Cosmos DB
```

> #### Tools
> - [Azure Cosmos DB Data Migration Tool (DMT)](https://github.com/AzureCosmosDB/data-migration-desktop-tool)

Reviewer: ```suggestion
#### Tools

- [Azure Cosmos DB Data Migration Tool (DMT)](https://github.com/AzureCosmosDB/data-migration-desktop-tool)
```

> #### Sample Migration Configuration (Cosmos DB to Blob Storage)

Reviewer: can we add a description about the json below? ie. how is it supposed to be used?

```json
{
  "Source": "Cosmos-nosql",
  "Sink": "Json-AzureBlob",
  "SourceSettings": {
    "ConnectionString": "<cosmos-connection-string>",
    "Database": "<cosmos-database-name>",
    "Container": "<cosmos-container-name>",
    "IncludeMetadataFields": true
  },
  "SinkSettings": {
    "AccountEndpoint": "https://<storage-account-name>.blob.core.windows.net",
    "ContainerName": "cosmosdb",
    "BlobName": "<cosmos-container-name>",
    "UseRbacAuth": true
  }
}
```

> #### Restore Configuration (Blob Storage to Cosmos DB)

Reviewer: ditto. can we add an example command line?

```json
{
  "Source": "Json-AzureBlob",
  "Sink": "Cosmos-nosql",
  "SourceSettings": {
    "AccountEndpoint": "https://<storage-account-name>.blob.core.windows.net",
    "ContainerName": "cosmosdb",
    "BlobName": "<cosmos-container-name>",
    "UseRbacAuth": true
  },
  "SinkSettings": {
    "ConnectionString": "<cosmos-connection-string>",
    "Database": "<cosmos-database-name>",
    "Container": "<cosmos-container-name>",
    "PartitionKeyPath": "/id"
  }
}
```

> ### 2. Migrating from Another Storage Account

Reviewer: ```suggestion
### Migrating from Another Storage Account
```

> #### Blob Storage Migration
> - Preferred Method: Object Replication

Reviewer: ```suggestion
#### Blob Storage Migration

Preferred Method: Object Replication
```

> > **Caution:** Cannot create cascading object replication policies on the same containers

Reviewer: ```suggestion
:::warning 

You can't create cascading object replication policies on the same container.

:::
```

this hint is not very clear anyway.

> ##### Terraform Object Replication Configuration

Reviewer: can we add a description about the code below?
